### ISSUE TITLE
Keyboard footprint merge discussion

### ISSUE THREAD

**Issue by perigoso** [KiCad/kicad-footprints#2416]
I'm creating this issue to have a centralized place for discussion regarding the massive keyboard footprints merge that i now split into multiple PRs.

This way we have somewhere to discuss things that affect all the fp's and we can better coordinate things.

The original PR #2394

**Comment by perigoso** [comment]
The original `Button_Switch_Keyboard.pretty` Includes a README.md, is this required? What's it's purpose? It seems most footprint libs don't include this file.

**Comment by diegoherranz** [comment]
> The original `Button_Switch_Keyboard.pretty` Includes a `README.md`, is this required? What's it's purpose? It seems most footprint libs don't include this file.

No. In fact, I would remove it. It may come from the old repo structure in which every .pretty was a separate repo.

**Comment by perigoso** [comment]
In that case maybe we'd remove it from all libs that include one? can I submit a PR with that or Just for the `Button_Switch_Keyboard.pretty` lib, since it would be the last file in that folder after the keyboard switch PRs move things around.

**Comment by perigoso** [comment]
So I had `_NoSilk` variants of footprints, because some people, including me, don't include silkscreen on the switch footprints for aesthetic reasons, but i was unaware that it was possible to hide it in bulk in `pcbnew`, so after some digging around and trying things out i came to this:

![image](https://user-images.githubusercontent.com/39195157/91366888-e3edaa80-e7fc-11ea-8244-fff18b60cee2.png)

This is a good versatile solution, and i feel the `_NoSilk` variants don't make sense given this option, maybe it might not be immediately clear to people but still, any input?

**Comment by evanshultz** [comment]
We require assets to comply with http://kicad-pcb.org/libraries/klc/, unless there's a specific reason not to or that doc needs an update. In this case, it's not. So yes, please only keep versions with silk as those do comply with the library guidelines.

**Comment by perigoso** [comment]
I am familiar with [KLC](http://kicad-pcb.org/libraries/klc/), I see no reason the `_NoSilk` versions of the footprints do not comply, hence why i was ok with including them in the first place.

**Comment by chschlue** [comment]
[Thou shalt not leave out silk](https://kicad-pcb.org/libraries/klc/F5.1/)

**Comment by perigoso** [comment]
The reference was still included, they just didn't have the square outline

**Comment by chschlue** [comment]
Yes, if you want to be really, really finnicky about the wording, you could argue that outline silk is not explicitly required.

But a silk pin 1 designator sure is. Please also add outline silk while your at it ;)

**Comment by perigoso** [comment]
I was not trying to word things my way, i was genuinely under the impression it was not required TIL i guess. I knew about the pin 1 marker but these footprints are not polarized, and they can only be mounted one way anyway.

I believe the hybrid footprints don't have a silkscreen outline, so I'll add it to those too.

**Comment by chschlue** [comment]
Sry, got you wrong there.

But as you said, users can always disable it if they don't like it.

**Comment by perigoso** [comment]
> In that case maybe we'd remove it from all libs that include one? can I submit a PR with that or Just for the `Button_Switch_Keyboard.pretty` lib, since it would be the last file in that folder after the keyboard switch PRs move things around.

What about this? the question got buried.

**Comment by chschlue** [comment]
Please stick to the switches for now.
Some amount of restructuring for v6 is coming and I would like some structure in "housekeeping" PRs like that if possible.

**Comment by perigoso** [comment]
Thinking these footprints are a good candidate for a script with [kicad-footprint-generator](https://github.com/pointhi/kicad-footprint-generator), should've come to this realization before having done all this manual work

**Comment by perigoso** [comment]
Created Scripts for the creation of these footprints

**Comment by perigoso** [comment]
Created a PR pointhi/kicad-footprint-generator#605 with the scripts to generate keyboard switch footprints, the PRs have been updated with the generated footprints, for reference here are the footprints:

![image](https://user-images.githubusercontent.com/39195157/92202993-1ab86600-ee78-11ea-9ed2-67838e06504b.png)
![image](https://user-images.githubusercontent.com/39195157/92204590-b5ff0a80-ee7b-11ea-9883-b09654cd7353.png)
![image](https://user-images.githubusercontent.com/39195157/92203039-2efc6300-ee78-11ea-83ea-b30654d52235.png)
![image](https://user-images.githubusercontent.com/39195157/92203055-36bc0780-ee78-11ea-9bfa-f2a4cbb3be80.png)
![image](https://user-images.githubusercontent.com/39195157/92203081-3e7bac00-ee78-11ea-8375-13989af58503.png)
![image](https://user-images.githubusercontent.com/39195157/92203093-46d3e700-ee78-11ea-8920-9aec6a695950.png)

**Comment by perigoso** [comment]
Of note, I removed the stabilizer mounting holes from these footprints, i intend to add them in separate footprints.

**Comment by perigoso** [comment]
There are some issues on the scripts that need to be fixed, namely 3D model settings, and reference designation.

**Comment by perigoso** [comment]
Some issues on the script have been fixed and the footprints re-generated.
